src/clients/Deluge.ts

```diff
@@ -22,11 +22,22 @@ enum DelugeErrorCode {
 	RPC_FAIL = 4,
 	BAD_JSON = 5,
 }
-interface TorrentInfo {
-	complete: boolean;
-	save_path: string;
-	label?: string;
-}
+type injectData = [
+	string,
+	string,
+	{
+		add_paused: boolean;
+		seed_mode: boolean;
+		download_location: string;
+	}
+];
+
+type DelugeJSON<ResultType> = {
+	result?: ResultType;
+	error?: ErrorType;
+};
+type ErrorType = { message?: string; code?: DelugeErrorCode };
+type TorrentStatus = { torrents?: Record<string, TorrentInfo> };
```

Comment:
it's technically not a problem but I prefer to define things before using them. In this case, ErrorType is used on line 37 before it's defined on line 39